### PR TITLE
[RLlib] Replace ordinary pygame imports by try_imports

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
Signed-off-by: Artur Niederfahrenhorst <artur@anyscale.com>

## Why are these changes needed?

Replaces imports of pygame with try_imports, akin to the ones we use for frameworks.
This helps us output a more informative import error.

## Related issue number

https://github.com/ray-project/ray/issues/31321

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/env/wrappers/open_spiel.py [hunk 1]
```diff
@@ -1,9 +1,12 @@
-from gymnasium.spaces import Box, Discrete
-import numpy as np
-import pyspiel
 from typing import Optional
 
+import numpy as np
```
Comment (sven1977) [inline rllib/env/wrappers/open_spiel.py:3]: Just curious, is this a new sorting convention? Three blocks: a) included python modules, b) other third-party imports, c) ray imports?

File: rllib/env/utils.py [hunk 2]
```diff
@@ -2,6 +2,34 @@
...
+
+@PublicAPI
+def try_import_pyspiel(error: bool = False):
+    """Tries importing pyspiel and returns the module (or None).
+
+    Args:
+        error: Whether to raise an error if pyspiel cannot be imported.
+
+    Returns:
+        The tfp module.
```
Comment (sven1977) [inline rllib/env/utils.py:16]: -> "pyspiel", not tfp

File: rllib/examples/self_play_league_based_with_open_spiel.py [hunk 3]
```diff
@@ -31,22 +31,25 @@
...
 
 import argparse
-import numpy as np
 import os
-from open_spiel.python.rl_environment import Environment
-import pyspiel
 import re
 
+import numpy as np
+from open_spiel.python.rl_environment import Environment
```
Comment (sven1977) [inline rllib/examples/self_play_league_based_with_open_spiel.py:38]: Dumb question: what was the difference between `open_spiel` and `pyspiel` again? Would we need to return the `open_spiel` package as well from `try_import_pyspiel()`?

File: rllib/examples/self_play_with_open_spiel.py [hunk 4]
```diff
@@ -19,21 +19,24 @@
...
 
 import argparse
-import numpy as np
 import os
-import pyspiel
-from open_spiel.python.rl_environment import Environment
 import sys
 
+import numpy as np
+from open_spiel.python.rl_environment import Environment
```
Comment (sven1977) [inline rllib/examples/self_play_with_open_spiel.py:26]: same question as above.

**Review by sven1977 — COMMENTED** [review]
Thanks for the fixes. Just a few questions on pyspiel vs open_spiel.

File: rllib/env/wrappers/open_spiel.py (hunk 1 above)
Comment (ArturNiederfahrenhorst) [reply rllib/env/wrappers/open_spiel.py:3]: The three categories you describe (and the formating changes here) conform with PEP-8 I belive.
-> https://peps.python.org/pep-0008/#imports
I think we should adhere at least to PEP-8. The google style guide is optional.

I use PyCharm's "optimize imports" feature to do these, but PyCharm also groups "import x as y" vs "import x" and separates them which makes for apparent missing alphabetical order.

File: rllib/examples/self_play_league_based_with_open_spiel.py (hunk 3 above)
Comment (ArturNiederfahrenhorst) [reply rllib/examples/self_play_league_based_with_open_spiel.py:38]: open spiel uses pygame, which is a game engine with a python interface.

Comment (ArturNiederfahrenhorst) [reply rllib/examples/self_play_league_based_with_open_spiel.py:38]: They are separate packages though.
We list pygame in our test requirements and open_spiel nowhere.

Comment (ArturNiederfahrenhorst) [reply rllib/examples/self_play_league_based_with_open_spiel.py]: I have added open_spiel to this PR with the same behaviour!

**Review by sven1977 — APPROVED** [review]
LGTM.

**Pull request merged**